src/VisualStudio/Core/Def/Implementation/Debugging/AbstractBreakpointResolver.cs

```diff
@@ -87,59 +84,72 @@ private BreakpointResolutionResult CreateBreakpoint(ISymbol methodSymbol)
-            //         the same way as any other property.
-            //         Destructors in C# can be resolved using the method name "Finalize". The resulting string
-            //         representation will use C# language format ("C.~C()").  I verified that this works with
-            //         "Break at Function" (breakpoint is correctly set and can be hit), so I don't see a reason
-            //         to prohibit this (even though the old language service didn't support it).
-            var members = await FindMembersAsync(nameParts, cancellationToken).ConfigureAwait(false);
-
-            // Filter down the list of symbols to "applicable methods", specifically:
-            // - "regular" methods
-            // - constructors
-            // - destructors
-            // - properties
-            // - operators?
-            // - conversions?
-            // where "applicable" means that the method or property represents a valid place to set a breakpoint
-            // and that it has the expected number of parameters
-            return members.Where(m => IsApplicable(m, parameterCount, cancellationToken));
+            try
+            {
+                ParseText(out var nameParts, out var parameterCount);
+
+                // Notes:  In C#, indexers can't be resolved by any name.  This is acceptable, because the old language
+                //         service wasn't able to resolve them either.  In VB, parameterized properties will work in
+                //         the same way as any other property.
+                //         Destructors in C# can be resolved using the method name "Finalize". The resulting string
+                //         representation will use C# language format ("C.~C()").  I verified that this works with
+                //         "Break at Function" (breakpoint is correctly set and can be hit), so I don't see a reason
+                //         to prohibit this (even though the old language service didn't support it).
+                var members = await FindMembersAsync(nameParts, cancellationToken).ConfigureAwait(false);
+
+                // Filter down the list of symbols to "applicable methods", specifically:
+                // - "regular" methods
+                // - constructors
+                // - destructors
+                // - properties
+                // - operators?
+                // - conversions?
+                // where "applicable" means that the method or property represents a valid place to set a breakpoint
+                // and that it has the expected number of parameters
+                return members.Where(m => IsApplicable(m, parameterCount, cancellationToken)).
+                    Select(CreateBreakpoint).ToImmutableArrayOrEmpty();
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+            {
+                return ImmutableArray<BreakpointResolutionResult>.Empty;
```

Comment:
Also, unless we crash in the filter we'd need to mark the exception with some flag to avoid reporting NFW higher on the stack again, if we rethrow it.